### ISSUE TITLE
Error: Internal error: fetch failed when try to login

### ISSUE THREAD

**Issue by digitalfantazy** [awinogrodzki/next-firebase-auth-edge#249]
Hello everyone, maybe someone has encountered this error, I don't understand why it occurs. Although I can work normally for some time and everything is ok, but then this error appears. I would be very grateful for an answer, I have been struggling with it for a very long time(

Here is this error 

 ⨯ Error: Internal error: fetch failed
    at FirebaseTokenVerifier.mapJoseErrorToAuthError (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:79:16)
    at eval (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:64:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async FirebaseTokenVerifier.decodeAndVerify (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:28:9)
    at async FirebaseTokenVerifier.verifyJWT (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/token-verifier.js:21:25)
    at async verifyIdToken (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/index.js:192:32)
    at async getCustomIdAndRefreshTokens (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/auth/index.js:223:30)
    at async setAuthCookies (webpack-internal:///(middleware)/./node_modules/next-firebase-auth-edge/lib/next/cookies.js:132:26)
    at async eval (webpack-internal:///(middleware)/./app/services/middlewares/authMiddleware.ts:24:26)
    at async adapter (webpack-internal:///(middleware)/./node_modules/next/dist/esm/server/web/adapter.js:179:16)
    
    
    This is my code 
```
import { NextFetchEvent, NextRequest, NextResponse } from 'next/server';
import { authConfig } from '@/config/authConfig';
import { authMiddleware, redirectToHome, redirectToLogin } from 'next-firebase-auth-edge';
import { CustomMiddleware } from './chain';

const PUBLIC_PATHS = ['/', '/history', '/graphql', '/rest'];

const AUTH_PATHS = ['/sign-up', '/sign-in'];

export default function AuthorizationMiddleware(middleware: CustomMiddleware) {
  return async (request: NextRequest, event: NextFetchEvent) => {
    const currentPath = request.nextUrl.pathname;

    const response = await authMiddleware(request, {
      loginPath: '/en/api/login',
      logoutPath: '/en/api/logout',
      enableMultipleCookies: authConfig.enableMultipleCookies,
      apiKey: authConfig.apiKey,
      cookieName: authConfig.cookieName,
      cookieSignatureKeys: authConfig.cookieSignatureKeys,
      cookieSerializeOptions: authConfig.cookieSerializeOptions,
      serviceAccount: authConfig.serviceAccount,

      handleValidToken: ({ decodedToken }, headers) => {
        if (AUTH_PATHS.includes(currentPath) && decodedToken) {
          return Promise.resolve(redirectToHome(request));
        }
        return Promise.resolve(
          NextResponse.next({
            request: {
              headers,
            },
          })
        );
      },
      handleInvalidToken: (reason) => {
        console.info('Missing or malformed credentials', { reason });
        const locale = currentPath.split('/')[1];
        const loginPath = `/${locale}/sign-in`;
        console.log(loginPath);

        if (PUBLIC_PATHS.includes(currentPath)) {
          console.log(!PUBLIC_PATHS.includes(currentPath));
          const url = new URL(loginPath, request.url);
          return Promise.resolve(NextResponse.redirect(url.toString()));
        }

        return Promise.resolve(NextResponse.next());
      },
      handleError: (error) => {
        console.error('Unhandled authentication error', { error });
        return Promise.resolve(
          redirectToLogin(request, {
            path: '/sign-in',
            publicPaths: PUBLIC_PATHS,
          })
        );
      },
    });
    return middleware(request, event, response);
  };
}
```
  
  
```
import chain from './app/services/middlewares/chain';
import AuthorizationMiddleware from './app/services/middlewares/authMiddleware';
import i18Middleware from './app/services/middlewares/i18Middleware';

const middlewares = [i18Middleware, AuthorizationMiddleware];
export default chain(middlewares);

export const config = {
  matcher: ['/((?!static|.*\\..*|_next).*)'],
};

```

**Comment by awinogrodzki** [comment]
Hey @digitalfantazy!

It seems that the following `fetch` call fails: https://github.com/awinogrodzki/next-firebase-auth-edge/blob/main/src/auth/signature-verifier.ts#L70

This call is relatively easy to debug. In this case fetch is always called with `https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com` URL.

I think you're experiencing a similar issue to this: https://stackoverflow.com/questions/74165121/next-js-fetch-request-gives-error-typeerror-fetch-failed

Basically, for some reason a call to `https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com` fails before a server has a chance to respond. 

Do you think there might be some periodical connection problem on the server that you're operating? 

To dig in, could you provide following information:

What is the system the server is running on (where error happens)? Ideally, could you provide the output of `npx next info` command? I am interested in OS and processor type above all

What is `next-firebase-auth-edge` and Next.js version you're using?

**Comment by digitalfantazy** [comment]
Here is the output of the command npx next info

![image](https://github.com/user-attachments/assets/966ff0f8-1d2b-4901-9f15-cd750b3a8d49)

Here is the versions Next.js and next-firebase-auth-edge 
    "next": "14.2.5",
    "next-firebase-auth-edge": "^1.7.0-canary.2",
    
Speaking about periodic problems with connection on the server, I think that this is unlikely, I do not use VPN or any proxies, if we are talking about them

**Comment by awinogrodzki** [comment]
@digitalfantazy thanks for details!

I've just released `next-firebase-auth-edge@1.7.0-canary.13` with improvements around error handling.

In short: Fetch Error should contain `cause` property that describes the error that happens inside Next.js fetch polyfill: https://stackoverflow.com/questions/78364079/error-while-running-next-js-project-typeerror-fetch-failed-and-unable-to-ve

`v1.7.0-canary.13` improves `AuthError` by exposing `cause` and `stack` from original error.

Can I ask you to update to this version, try to reproduce the error and share the `stack` and `cause` property of the error? You can log it to console like so:

```ts
    handleError: async (error) => {
      console.log({ cause: error.cause, stack: error.stack });

      return redirectToLogin(request, {
        path: '/login',
        publicPaths: PUBLIC_PATHS
      });
    }
```
